lib/cucumber/rails.rb

```diff
@@ -1,4 +1,7 @@
-env_caller = File.dirname(caller.detect { |f| f =~ /\/env\.rb:/ }) if caller.detect { |f| f =~ /\/env\.rb:/ }
+if caller.detect { |f| f =~ /\/env\.rb:/ }
+  env_caller = File.dirname(caller.detect { |f| f =~ /\/env\.rb:/ })
+end
```

Comment:
Perhaps this can assign to `env_caller` the result of the `if` expression?

Why I say that: Now, the `env_caller` after line 3 is undefined, if we did not visit line 2.

Reply:
It'll never be undefined, it'll be `nil` though, because of the way ruby lazy-evaluates local variables. I remember when I first learnt this it confused me.

```
a = 3
if a > 3
  b = 5
end

puts b => `nil` # This seems odd but it works.
```